### PR TITLE
[cli] Make chunk size configurable for chunked publish

### PR DESCRIPTION
**Pull request by 0xjunha**
## Description

Updated the chunked publish mode in the CLI to include an option for configuring the chunk size.

This gives more flexibility since the chunked publish mode utilizes nearly the maximum payload size permitted for a single transaction. This change helps to handle failures when gas charges are updated.

* Added a new `chunk_size` option to `ChunkedPublishOption`
* Updated `MAX_CHUNK_SIZE_IN_BYTES` to be of type `&str` with a value of `55000`

## How Has This Been Tested?

* e2e-move-test
* manually building & testing the CLI

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [x] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Move Compiler
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by wrwg — APPROVED** [review]

File: crates/aptos/src/common/types.rs [hunk 1]
```diff
@@ -2370,8 +2370,16 @@ pub struct ChunkedPublishOption {
...
     /// under move-examples/large_packages.
     #[clap(long, default_value = LARGE_PACKAGES_MODULE_ADDRESS, value_parser = crate::common::types::load_account_arg)]
     pub(crate) large_packages_module_address: AccountAddress,
+
+    /// Size of the code chunk in bytes for splitting bytecode and metadata of large packages
+    ///
+    /// By default, the chunk size is set to `CHUNK_SIZE_IN_BYTES`. A smaller chunk size will result
+    /// in more transactions required to publish a package, while a larger chunk size might cause
+    /// transaction to fail due to exceeding the execution gas limit.
+    #[clap(long, default_value = CHUNK_SIZE_IN_BYTES, value_parser = clap::value_parser!(usize))]
```
Comment (banool) [inline crates/aptos/src/common/types.rs:2383]: Try using just default_value_t, then you won't need the string default or the parsing function.

File: crates/aptos/CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ All notable changes to the Aptos CLI will be captured in this file. This project
 # Unreleased
 
 - Add `--node-api-key` flag to `aptos move replay` to allow for querying the fullnode with an API key.
-
+- Add `--chunk-size` flag to allow configuring chunk size for chunked publish mode.
```
Comment (banool) [inline crates/aptos/CHANGELOG.md:8]: You should also mention that you're lowering the default value from x to y.

**Review by banool — APPROVED** [review]

**Review by gregnazario — APPROVED** [review]
Looks fine, dport's comments should be addressed.  Changelog should be 60 -> 50 k

And the default_value_t will make it easier but it's not the end of the world

File: crates/aptos/src/common/types.rs (hunk 1 above)
Comment (0xjunha) [reply crates/aptos/src/common/types.rs]: Thanks for the suggestion! fixed it

File: crates/aptos/CHANGELOG.md (hunk 2 above)
Comment (0xjunha) [reply crates/aptos/CHANGELOG.md:8]: Fixed!

**Pull request merged**